code/_onclick/hud/radial.dm

```diff
@@ -313,31 +331,76 @@
 		qdel(custom_check)
 		custom_check = null
 	. = ..()
+
+/////////////////////////////////////////Recursive radial menu, currently used by Safes
+
+/datum/radial_menu/proc/do_it_again()
+	selected_choice = null
+	if (istype(anchor,/obj/structure/safe))
```

Comment:
This functionality should probably be moved into a radial_menu/safe subtype, rather than typechecking here (muh OOP)